basic-ruby/stylish-ruby/triangle_facts_refactored.rb

```diff
@@ -0,0 +1,74 @@
+# Killer facts about triangles AWW YEAH
+class Triangle
+  attr_accessor :side1, :side2, :side3
+
+  def initialize(side1,side2,side3)
+    @side1 = side1
+    @side2 = side2
+    @side3 = side3
+  end
+
+  def equilateral
+    return side1 == side2 && side2 == side3
+  end
+
+  def isosceles
+    return [side1,side2,side3].uniq.length == 2
+  end
+
+  def scalene
+    !(equilateral || isosceles) 
+  end
+
+  def recite_facts
+    puts "#{equilateral_fact}", 
+          "#{isosceles_fact}", 
+          "#{scalene_fact}", 
+          "#{angles_fact}",
+          "#{right_triangle_fact}"
+  end
+
+  def calculate_angles(a,b,c)
+    angleA = radians_to_degrees(Math.acos((b**2 + c**2 - a**2) / (2.0 * b * c)))
+    angleB = radians_to_degrees(Math.acos((a**2 + c**2 - b**2) / (2.0 * a * c)))
+    angleC = radians_to_degrees(Math.acos((a**2 + b**2 - c**2) / (2.0 * a * b)))
+
+    [angleA, angleB, angleC]
+  end
+
+  def radians_to_degrees(rads)
+    (rads * 180 / Math::PI).round
+  end
+
+  def equilateral_fact
+    'This triangle is equilateral!' if equilateral
```

Comment:
I don't really like this syntax for `if` statements, because it's easy to overlook the fact that there's even an `if` at the end of a long line. IMO, a better way to write this would be with a guard clause:

```ruby
def equilateral_fact
  return unless equilateral
  'This triangle is equilateral!'
end
```